numcodecs/bitround.py

```diff
@@ -0,0 +1,82 @@
+import numpy as np
+
+
+from .abc import Codec
+from .compat import ensure_ndarray_like, ndarray_copy
+
+max_bits = {
+    "float16": 10,
+    "float32": 23,
+    "float64": 52,
+}
```

Comment:
Should we have some comment (or module docstring) explaining what these are. The `max_bits` is less straightforward (at least to me). So a bit more detail on that would be helpful.